plugins/source/okta/codegen/recipes/recipes.go

```diff
@@ -0,0 +1,156 @@
+package recipes
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/caser"
+	"github.com/cloudquery/plugin-sdk/codegen"
+	"github.com/gertd/go-pluralize"
+)
+
+type Resource struct {
+	// Table is the table definition that will be used to generate the cloudquery table
+	Table *codegen.TableDefinition
+	// DataStruct that will be used to generate the cloudquery table
+	DataStruct interface{}
+	// SkipFields fields in go struct to skip when generating the table from the go struct
+	SkipFields []string
+	Template   string
+	Multiplex  string
+
+	ExtraColumns []codegen.ColumnDefinition
+	PKColumns    []string
+
+	PreResourceResolver   string
+	PostResourceResolver  string
+	Relations             []string
+	UnwrapEmbeddedStructs bool
+
+	Service string // Required
+
+	SkipServiceInTableName bool // Don't prepend service name to table name
+	SkipParentInTableName  bool // Don't prepend parent name to table name
+
+	// These are inferred with reflection but can be overridden
+	SubService string // Inferred from DataStruct name
+	TableName  string // singular Service + plural SubService
+
+	// These are auto calculated
+	ImportClient     bool   // true if the resource/column resolvers use the client package
+	Filename         string // Calculated from TableName
+	TableFuncName    string // Calculated from TableName
+	ResolverFuncName string // Calculated from TableFuncName
+
+	// Used for generating better table names
+	parent   *Resource
+	children []*Resource
+}
+
+var (
+	pluralizeClient *pluralize.Client
+	csr             *caser.Caser
+)
+
+func init() {
+	pluralizeClient = pluralize.NewClient()
+	csr = caser.New()
+}
+
+func (r *Resource) Infer() {
+	// Set defaults and/or infer fields
+	if r.Template == "" {
+		r.Template = "resource"
+	}
+
+	ds := reflect.TypeOf(r.DataStruct)
+	if ds.Kind() == reflect.Ptr {
+		ds = ds.Elem()
+	}
+
+	if r.Service == "" {
+		log.Fatalf("Service is required for %s", r.SubService)
+	}
+	if r.SubService == "" {
+		r.SubService = csr.ToSnake(pluralizeClient.Singular(ds.Name()))
+	}
+}
+
+func (r *Resource) GenerateNames() {
```

Comment:
Maybe those should be set manually rather then a complex logic that generates them?

Reply:
As you can see in the next line there's a name override option.